src/awkward/_v2/_connect/rdataframe/to_rdataframe.py

```diff
@@ -6,8 +6,20 @@
 import awkward._v2._connect.cling  # noqa: E402
 
 import ROOT
+import threading
 
 compiler = ROOT.gInterpreter.Declare
+compiler_lock = threading.Lock()
+compiler(
+    """
+#include <Python.h>
+"""
+)
+
+
+def compile(source_code):
+    with compiler_lock:
+        return compiler(source_code)
```

Comment:
Might as well be consistent and use the lock for the `#include <Python.h>` as well as everything else.

Also, no reason to introduce the name "`compiler`" distinct from "`compile`". It's not used anywhere else.

```suggestion
def compile(source_code):
    with compiler_lock:
        return ROOT.gInterpreter.Declare(source_code)


compiler_lock = threading.Lock()
compiler("#include <Python.h>\n")
```

Reply:
I was wrong: there's one reference to the original "`compiler`", rather than "`compile`", which shows that this clean-up is helpful:

https://github.com/scikit-hep/awkward-1.0/blob/19f92b50bf4364cecd8ca647caf59297cc3b6572/src/awkward/_v2/_connect/rdataframe/to_rdataframe.py#L62

That line would have to be changed in addition to this, and GitHub won't let me make it a one-click commit.

Reply:
done